content/200-concepts/100-components/02-prisma-client/260-client-extensions/120-query.mdx

```diff
@@ -1,16 +1,176 @@
+- [A specific operation in all models of your schema](#add-a-custom-query-to-a-specific-operation-in-=all-models-of-your-schema)
+- [All operations in a specific model of your schema](#add-a-custom-query-to-all-operations-in-a-specific-model)
+- [All operations in all models of your schema](#add-a-custom-query-to-all-operations-in-all-models-of-your-schema)
+
+You can add a query to a specific operation in a model, to a specific operation to all operations in a model, or to all operations in all models of your schema.
+
+To create a custom query, use the following structure:
+
+```ts
+const xprisma = prisma.$extends({
+  name?: 'name',
+  query?: {
+    user: { ... } //in this case, we add a query to the `user` model
+  },
+});
+```
+
+The properties are as follows:
+
+- `name`: (optional) specifies a name for the extension that appears in error logs.
+- `query`: (optional) defines a custom query.
+
+### Add a custom query to a specific operation in a specific model
+
+The `query` object contains functions that map to the names of the [Prisma Client operations](/docs/reference/api-reference/prisma-client-reference#model-queries), such as `findUnique`, `findFirst`, `findMany`, `count`, and `create`. The following example defines a custom query that finds users who are older than 18 years:
+
+```ts
+const xprisma = prisma.$extends({
+  query: {
+    user: {
+      async findMany({ model, action, args, data }) {
+        // `where` can be `undefined`, so `args.where.age` might
+        // fail. To work around this, we assign an object and
+        // backfill the rest of the `where` clause:
+        args.where = { age: { gt: 18 }, ...args.where }
+        const users = await data
+        return data
+      },
+    },
+  },
+})
+
+await xprisma.user.findMany() // returns users whose age is greater than 18
+```
+
+In the above example, a call to `xprisma.user.findMany` triggers `query.user.findMany`. Each callback receives a type-safe `{ model, operation, args, data }` object that describes the query. This object has the following properties:
+
+- `model`: the name of the containing model for the query that we want to extend.
+
+  In the above example, the `model` is a string of type `"User"`.
+
+- `operation`: the name of the operation being extended and executed.
+
+  In the above example, the `operation` is a string of type `"findMany"`.
+
+- `args`: the specific query input information to be extended.
+
+  This is a type-safe object that you can mutate before the query happens. You can mutate any of the properties in `args`. Exception: you cannot mutate `include` or `select` because that would change the expected output type and break type safety.
+
+- `data`: a promise for the result of the query.
+
+  - You can use `await` and then mutate this promise, because its value is type-safe. TypeScript catches any unsafe mutations on the object.
```

Comment:
```suggestion
  - You can use `await` and then mutate the result of this promise, because its value is type-safe. TypeScript catches any unsafe mutations on the object.
```